### PR TITLE
chat: mark chat channel read when unread divider is visible on screen

### PR DESCRIPTION
**Pull request by rcrdlbl**
see: #600.

### CODE REVIEW

**Review by nerveharp — APPROVED** [review]
Gonna steal this one for private/future use, seems nice. Thanks!

File: ui/src/chat/ChatMessage/DateDivider.tsx [hunk 1]
```diff
@@ -1,17 +1,23 @@
...
   date: Date;
   unreadCount?: number;
+  viewRef?: (node?: Element | null | undefined) => void;
 }
 
-export default function DateDivider({ date, unreadCount }: DateDividerProps) {
+export default function DateDivider({
+  date,
+  unreadCount,
+  viewRef,
```
Comment (arthyn) [inline ui/src/chat/ChatMessage/DateDivider.tsx:15]: This really should use `React.forwardRef` instead

Comment (rcrdlbl) [reply ui/src/chat/ChatMessage/DateDivider.tsx:15]: the library *might* not play well with it – but i'll poke around and see

Comment (rcrdlbl) [reply ui/src/chat/ChatMessage/DateDivider.tsx:15]: Okay, just pushed.

**Review by arthyn — APPROVED** [review]
lgtm!

**Pull request merged**